platform/linux-generic/odp_classification.c

```diff
@@ -856,23 +856,28 @@ static inline int verify_pmr_packet_len(odp_packet_hdr_t *pkt_hdr,
 	return 0;
 }
 
-static inline int verify_pmr_ip_proto(const uint8_t *pkt_addr,
-				      odp_packet_hdr_t *pkt_hdr,
-				      pmr_term_value_t *term_value)
+static inline int verify_pmr_ipv4_proto(const _odp_ipv4hdr_t *ipv4, pmr_term_value_t *term_value)
 {
-	const _odp_ipv4hdr_t *ip;
 	uint8_t proto;
 
-	if (!pkt_hdr->p.input_flags.ipv4)
-		return 0;
-	ip = (const _odp_ipv4hdr_t *)(pkt_addr + pkt_hdr->p.l3_offset);
-	proto = ip->proto;
+	proto = ipv4->proto;
 	if (term_value->match.value == (proto & term_value->match.mask))
 		return 1;
 
 	return 0;
 }
 
+static inline int verify_pmr_ipv6_next_hdr(const _odp_ipv6hdr_t *ipv6, pmr_term_value_t *term_value)
+{
+	uint8_t next_hdr;
+
+	next_hdr = ipv6->next_hdr;
+	if (term_value->match.value == (next_hdr & term_value->match.mask))
```

Comment:
This same pattern is used for all other PMRs. So, I'd maintain the current pattern, and change the pattern for all PMRs simultaneously in a non-API changing PR.

Since match.value and match.mask are uint64_t, we might want to change the match target variable (next_hdr) to uint64_t as well.

Reply:
Yes, fair point.